src/ngraph_encapsulate_op.cc

```diff
@@ -195,6 +215,16 @@ class NGraphEncapsulateOp : public tf::OpKernel {
       }
       last_used_src_ptrs[i] = src_ptr;
       ng_inputs.push_back(t);
+      };
+
+      if(m_ng_backend_name == "NNP") {
+      auto t = m_ng_backend->create_tensor(ng_element_type, ng_shape);
```

Comment:
For non-host backends, we shall only be calling `m_ng_backend->create_tensor(ng_element_type, ng_shape);` for the first time. Then we shall use the cached version of the created tensor. `m_ng_backend->create_tensor` is not free for non-CPU backends. We can afford to create tensors multiple times in CPU because the `src_ptr` is reused when we call `m_ng_backend->create_tensor(ng_element_type, ng_shape, src_ptr);`

We shall have a map of `<ng::Function, vector<ng::runtime::TensorView>> map_ng_function_to_inputs` to cache the input tensors.

Reply:
This seems like something that should be a part of Core since we are doing the same at both ends 🤔